src/tree/map.rs

```diff
@@ -329,12 +324,7 @@ impl<K, V, C> TreeMap<K, V, C> where C: Compare<K> {
     /// ```
     #[unstable = "matches collection reform specification, waiting for dust to settle"]
     pub fn iter_mut(&mut self) -> IterMut<K, V> {
-        IterMut {
-            stack: vec![],
-            node: deref_mut(&mut self.root),
-            remaining_min: self.length,
-            remaining_max: self.length
-        }
+        IterMut { iter: node::Iter::new(node::as_ref(&self.root), self.len) }
```

Comment:
Alas, without parametric mutability, we resort to using `transmute` in the `Iterator` impl itself. I'm not sure if we need to include a field like `PhantomData<&mut V>` in the `IterMut` definition to make sure everything's OK.